localstack-core/localstack/services/sns/v2/provider.py

```diff
@@ -1,9 +1,158 @@
+        )
+        topic: Topic = store.topics.get(topic_ARN)
+        attributes = attributes or {}
+        if topic:
+            attrs = topic.attributes
+            for k, v in attributes.values():
+                if not attrs.get(k) or not attrs.get(k) == v:
+                    # TODO:
+                    raise InvalidParameterException("Fix this Exception message and type")
+            return CreateTopicResponse(TopicArn=topic_ARN)
+
+        topic = Topic(name=name, attributes=attributes, arn=topic_ARN, context=context)
+
+        attributes = attributes or {}
+        if attributes.get("FifoTopic") and attributes["FifoTopic"].lower() == "true":
+            fifo_match = re.match(SNS_TOPIC_NAME_PATTERN_FIFO, name)
+            if not fifo_match:
+                # TODO: check this with a separate test
+                raise InvalidParameterException(
+                    "Fifo Topic names must end with .fifo and must be made up of only uppercase and lowercase ASCII letters, numbers, underscores, and hyphens, and must be between 1 and 256 characters long."
+                )
+        else:
+            # AWS does not seem to save explicit settings of fifo = false
+            try:
+                if topic.attributes["FifoTopic"]:
+                    del topic.attributes["FifoTopic"]
+            except KeyError:
+                pass
+            name_match = re.match(SNS_TOPIC_NAME_PATTERN, name)
+            if not name_match:
+                raise InvalidParameterException("Invalid parameter: Topic Name")
+
+        # if attributes:
+        # self.set_topic_defaults(topic, context)
+        # topic["attributes"] = attributes
+        store.topics[topic_ARN] = topic
+        # todo: tags
+
+        return CreateTopicResponse(TopicArn=topic_ARN)
+
+    def get_topic_attributes(
+        self, context: RequestContext, topic_arn: topicARN, **kwargs
+    ) -> GetTopicAttributesResponse:
+        topic: Topic = self._get_topic(arn=topic_arn, context=context)
+        if topic:
+            attributes = topic.attributes
+            return GetTopicAttributesResponse(Attributes=attributes)
+        else:
+            raise NotFoundException("Topic does not exist")
+
+    def delete_topic(self, context: RequestContext, topic_arn: topicARN, **kwargs) -> None:
+        store = self.get_store(context.account_id, context.region)
+
+        if store.topics.get(topic_arn):
+            del store.topics[topic_arn]
+
+    def list_topics(
+        self, context: RequestContext, next_token: nextToken | None = None, **kwargs
+    ) -> ListTopicsResponse:
+        store = self.get_store(context.account_id, context.region)
+        topics = {"Topics": [{"TopicArn": t.arn} for t in store.topics.values()]}
+        topics = PaginatedList(topics["Topics"])
+        page, nxt = topics.get_page(
+            lambda topic: topic["TopicArn"], next_token=next_token, page_size=100
```

Comment:
nit: using the topic ARN as the pagination token might be a bit of an issue (some client might not handle the arn format for that), but this can left as is for now, just a thought